test/irmin/test_tree.ml

```diff
@@ -542,6 +543,58 @@ let test_fold_force _ () =
       contents
   in
 
+  (* Ensure that [fold ~force:`True ~cache:false] visits newly added values. *)
+  let* () =
+    let* () = clear_and_assert_lazy sample_tree in
+    Tree.add sample_tree [ "a"; "ad" ] "v-ad" >>= fun updated_tree ->
+    let visited = ref false in
+    let contents k v () =
+      if equal_key k [ "a"; "ad" ] then (
+        visited := true;
+        if not (String.equal v "v-ad") then
+          Alcotest.failf "Newly added contents at %a not visited during fold"
+            pp_key k);
```

Comment:
I don't understand the purpose of that check.

Reply:
removed that part of the test